projects/ngneat/hotkeys/src/lib/hotkeys.service.ts

```diff
@@ -0,0 +1,73 @@
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable } from '@angular/core';
+import { EventManager } from '@angular/platform-browser';
+import { Observable } from 'rxjs';
+
+interface Options {
+  element: HTMLElement;
+  trigger: 'keydown' | 'keyup';
+  description: string;
+  showInHelp: boolean;
+  group: string;
+}
+
+export type Hotkey = Partial<Options> & { keys: string };
+
+@Injectable({
+  providedIn: 'root'
+})
+export class HotkeysService {
+  private readonly hotkeys = new Map();
+  private readonly defaults: Options = {
+    trigger: 'keydown',
+    element: this.document.documentElement,
+    description: '',
+    group: '',
+    showInHelp: true
+  };
+
+  constructor(private eventManager: EventManager, @Inject(DOCUMENT) private document: Document) {}
+
+  getShortcutsGroups(): string[] {
+    // map to get group name and remove duplicates
+    return Array.from(this.hotkeys)
+      .map(m => m[1])
+      .map(h => h.group)
+      .filter((el, idx, array) => array.indexOf(el) === idx);
```

Comment:
`(el, idx, array) => array.indexOf(el) === idx` - move to a pure function, and use it in both places.

Reply:
Also, I'm not sure how you'll have duplicates.

Reply:
Group is stored inside Hotkey, so if you add multiple hotkeys with the same group name, you’ll have duplicated group names.

Reply:
Why not make the structure simpler?
```ts
{
  [groupName]: hotkeys[]
}
```

Reply:
Sure!